src/Microsoft.OData.Core/UriParser/Resolver/UnqualifiedODataUriResolver.cs

```diff
@@ -46,28 +46,50 @@ public override IEnumerable<IEdmOperation> ResolveBoundOperations(IEdmModel mode
             {
                 return base.ResolveBoundOperations(model, identifier, bindingType);
             }
-
-            return FindAcrossModels<IEdmOperation>(model, identifier, this.EnableCaseInsensitive)
-                .Where(operation =>
-                    operation.IsBound
-                    && operation.Parameters.Any()
-                    && operation.HasEquivalentBindingType(bindingType));
+          
+            return GetBoundOperationsForModel(model, identifier, bindingType, this.EnableCaseInsensitive? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal);
         }
 
-        private static IEnumerable<T> FindAcrossModels<T>(IEdmModel model, String qualifiedName, bool caseInsensitive) where T : IEdmSchemaElement
+        private static IList<IEdmOperation> GetUnBoundOperationsForModel(IEdmModel model, string qualifiedName, StringComparison strComparison)
         {
-            Func<IEdmModel, IEnumerable<T>> finder = (refModel) =>
-                refModel.SchemaElements.OfType<T>()
-                .Where(e => string.Equals(qualifiedName, e.Name, caseInsensitive ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal));
+            IList<IEdmOperation> results = new List<IEdmOperation>();
 
-            IEnumerable<T> results = finder(model);
+            foreach (IEdmSchemaElement schemaElement in model.SchemaElements)
+            {
+                if (schemaElement is IEdmOperation operation && !operation.IsBound && string.Equals(qualifiedName, operation.Name, strComparison))
+                {
+                    results.Add(operation);
+                }
+            }
 
-            foreach (IEdmModel reference in model.ReferencedModels)
+            foreach(IEdmModel reference in model.ReferencedModels)
             {
-                results.Concat(finder(reference));
+                foreach (IEdmSchemaElement schemaElement in reference.SchemaElements)
+                {
+                    if (schemaElement is IEdmOperation operation && !operation.IsBound && string.Equals(qualifiedName, operation.Name, strComparison))
+                    {
+                        results.Add(operation);
+                    }
+                }
             }
 
             return results;
         }
+
+        private static IList<IEdmOperation> GetBoundOperationsForModel(IEdmModel model, string qualifiedName, IEdmType bindingType, StringComparison strComparison)

```

Comment:
>qualifiedName [](http://example.com/codeflow?start=95&length=13)

please rename to "operationName" to avoid confusion, since this is not the qualified name.